### PR TITLE
[Part-V] TF-2819 Calendar event maybe data source and repo

### PR DESCRIPTION
**Pull request by tddang-linagora**
### Issue
- #2819

### CODE REVIEW

File: lib/features/email/data/datasource/calendar_event_datasource.dart [hunk 1]
```diff
@@ -2,10 +2,16 @@
...
+import 'package:jmap_dart_client/jmap/mail/calendar/reply/calendar_event_accept_response.dart';
+import 'package:jmap_dart_client/jmap/mail/calendar/reply/calendar_event_maybe_response.dart';
 import 'package:tmail_ui_user/features/email/domain/model/event_action.dart';
 
 abstract class CalendarEventDataSource {
   Future<Map<Id, List<CalendarEvent>>> parse(AccountId accountId, Set<Id> blobIds);
 
   Future<List<EventAction>> getListEventAction(String emailContents);
+
+  Future<CalendarEventAcceptResponse> accept(AccountId accountId, Set<Id> blobIds);
```
Comment (dab246) [inline lib/features/email/data/datasource/calendar_event_datasource.dart:14]:
```suggestion
  Future<CalendarEventAcceptResponse> acceptEventInvitation(AccountId accountId, Set<Id> blobIds);
```

File: lib/features/email/data/datasource/calendar_event_datasource.dart [hunk 2]
```diff
@@ -2,10 +2,16 @@
...
+
+  Future<CalendarEventMaybeResponse> maybe(AccountId accountId, Set<Id> blobIds);
```
Comment (dab246) [inline lib/features/email/data/datasource/calendar_event_datasource.dart:16]:
```suggestion
  Future<CalendarEventAcceptResponse> maybeEventInvitation(AccountId accountId, Set<Id> blobIds);
```

File: lib/features/email/domain/repository/calendar_event_repository.dart [hunk 3]
```diff
@@ -2,10 +2,16 @@
...
 import 'package:tmail_ui_user/features/email/domain/model/event_action.dart';
 
 abstract class CalendarEventRepository {
   Future<Map<Id, List<CalendarEvent>>> parse(AccountId accountId, Set<Id> blobIds);
 
   Future<List<EventAction>> getListEventAction(String emailContents);
+
+  Future<CalendarEventAcceptResponse> accept(AccountId accountId, Set<Id> blobIds);
+
+  Future<CalendarEventMaybeResponse> maybe(AccountId accountId, Set<Id> blobIds);
```
Comment (dab246) [inline lib/features/email/domain/repository/calendar_event_repository.dart:16]: idem

File: test/features/email/data/repository/calendar_event_repository_impl_test.dart [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+@GenerateNiceMocks([MockSpec<CalendarEventDataSource>()])
+void main() {
+  final calendarEventNetworkDataSource = MockCalendarEventDataSource();
+  final calendarEventDataSource = {
+    DataSourceType.network: calendarEventNetworkDataSource};
+  final calendarEventRepository = CalendarEventRepositoryImpl(calendarEventDataSource);
+  final accountId = AccountId(Id('123'));
+  final blobId = Id('blobId');
+
+  group('calendar event accept repository', () {
```
Comment (dab246) [inline test/features/email/data/repository/calendar_event_repository_impl_test.dart:25]:
```suggestion
  group('calendar event accept repository test', () {
```

File: test/features/email/data/repository/calendar_event_repository_impl_test.dart [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        .thenThrow(NotAcceptableCalendarEventException());
+      
+      // assert
+      expect(
+        () => calendarEventRepository.accept(accountId, {blobId}),
+        throwsA(isA<NotAcceptableCalendarEventException>()));
+    });
+  });
+
+  group('calendar event maybe repository', () {
```
Comment (dab246) [inline test/features/email/data/repository/calendar_event_repository_impl_test.dart:55]:
```suggestion
  group('calendar event maybe repository test', () {
```

File: lib/features/email/data/datasource/calendar_event_datasource.dart (hunk 2 above)
Comment (tddang-linagora) [reply lib/features/email/data/datasource/calendar_event_datasource.dart]: https://github.com/linagora/tmail-flutter/pull/2820/commits/691302fef21b0bf5672c745a2031207b4b2b2f8c

File: lib/features/email/domain/repository/calendar_event_repository.dart (hunk 3 above)
Comment (tddang-linagora) [reply lib/features/email/domain/repository/calendar_event_repository.dart]: https://github.com/linagora/tmail-flutter/commit/57d9b3af88ea6576b4dd0c4abd647c0134fe1054 & https://github.com/linagora/tmail-flutter/commit/691302fef21b0bf5672c745a2031207b4b2b2f8c

File: test/features/email/data/repository/calendar_event_repository_impl_test.dart (hunk 4 above)
Comment (tddang-linagora) [reply test/features/email/data/repository/calendar_event_repository_impl_test.dart:25]: check https://github.com/linagora/tmail-flutter/pull/2815#discussion_r1574208778

File: test/features/email/data/repository/calendar_event_repository_impl_test.dart (hunk 5 above)
Comment (tddang-linagora) [reply test/features/email/data/repository/calendar_event_repository_impl_test.dart:55]: idem

**Review by hoangdat — APPROVED** [review]

**Pull request merged**